File: framework/web/src/test/java/top/bettercode/simpleframework/web/DataDicBean.java
package top.bettercode.simpleframework.web;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.math.BigDecimal;
import java.util.Arrays;
import java.util.List;
import java.util.Objects;
import top.bettercode.simpleframework.web.resolver.Cent;
import top.bettercode.simpleframework.web.serializer.annotation.JsonBigDecimal;
import top.bettercode.simpleframework.web.serializer.annotation.JsonUrl;

@JsonIgnoreProperties(ignoreUnknown = true)
public class DataDicBean {

  private BigDecimal number1;
  @JsonBigDecimal(scale = 3)
  private BigDecimal number2;
  @JsonBigDecimal(scale = 3, reduceFraction = true)
  private BigDecimal number22;
  @JsonBigDecimal(scale = 4, toPlainString = true,percent = true,reduceFraction = true)
  private BigDecimal number3;
  @JsonBigDecimal(scale = 4, toPlainString = true, reduceFraction = true)
  private BigDecimal number4;


  private String name;
  private String code;
  private Integer intCode;
  @Cent
  private Long price;
  @JsonUrl
  private String path;
  private String path1;
  private String desc;

  private List<String> paths;
  private String[] pathArray;


  public BigDecimal getNumber1() {
    return number1;
  }

  public void setNumber1(BigDecimal number1) {
    this.number1 = number1;
  }

  public BigDecimal getNumber2() {
    return number2;
  }

  public void setNumber2(BigDecimal number2) {
    this.number2 = number2;
  }

  public BigDecimal getNumber22() {
    return number22;
  }

  public void setNumber22(BigDecimal number22) {
    this.number22 = number22;
  }

  public BigDecimal getNumber3() {
    return number3;
  }

  public void setNumber3(BigDecimal number3) {
    this.number3 = number3;
  }


  public BigDecimal getNumber4() {
    return number4;
  }

  public void setNumber4(BigDecimal number4) {
    this.number4 = number4;
  }

  public String getPath1() {
    return path1;
  }

  public void setPath1(String path1) {
    this.path1 = path1;
  }

  public String[] getPathArray() {
    return pathArray;
  }

  public void setPathArray(String[] pathArray) {
    this.pathArray = pathArray;
  }

  public List<String> getPaths1() {
    return paths;
  }


  public String[] getPathArray1() {
    return pathArray;
  }

  public Long getPrice() {
    return price;
  }

  public void setPrice(Long price) {
    this.price = price;
  }

  public List<String> getPaths() {
    return paths;
  }

  public void setPaths(List<String> paths) {
    this.paths = paths;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }

  public Integer getIntCode() {
    return intCode;
  }

  public void setIntCode(Integer intCode) {
    this.intCode = intCode;
  }

  public String getCode() {
    return code;
  }

  public void setCode(String code) {
    this.code = code;
  }

  public String getDesc() {
    return desc;
  }

  public void setDesc(String desc) {
    this.desc = desc;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (!(o instanceof DataDicBean)) {
      return false;
    }
    DataDicBean that = (DataDicBean) o;
    return Objects.equals(number1, that.number1) && Objects.equals(number2,
        that.number2) && Objects.equals(number22, that.number22)
        && Objects.equals(number3, that.number3) && Objects.equals(number4,
        that.number4) && Objects.equals(name, that.name) && Objects.equals(code,
        that.code) && Objects.equals(intCode, that.intCode) && Objects.equals(
        price, that.price) && Objects.equals(path, that.path) && Objects.equals(
        path1, that.path1) && Objects.equals(desc, that.desc) && Objects.equals(
        paths, that.paths) && Arrays.equals(pathArray, that.pathArray);
  }

  @Override
  public int hashCode() {
    int result = Objects.hash(number1, number2, number22, number3, number4, name, code, intCode,
        price, path, path1, desc, paths);
    result = 31 * result + Arrays.hashCode(pathArray);
    return result;
  }

  @Override
  public String toString() {
    return "DataDicBean{" +
        "number1=" + number1 +
        ", number2=" + number2 +
        ", number22=" + number22 +
        ", number3=" + number3 +
        ", number4=" + number4 +
        ", name='" + name + '\'' +
        ", code='" + code + '\'' +
        ", intCode=" + intCode +
        ", price=" + price +
        ", path='" + path + '\'' +
        ", path1='" + path1 + '\'' +
        ", desc='" + desc + '\'' +
        ", paths=" + paths +
        ", pathArray=" + Arrays.toString(pathArray) +
        '}';
  }
}

File: web/src/test/kotlin/top/bettercode/summer/test/web/DataDicBean.kt
package top.bettercode.summer.test.web

import top.bettercode.summer.web.resolver.Unit
import top.bettercode.summer.web.serializer.annotation.JsonBigDecimal
import top.bettercode.summer.web.serializer.annotation.JsonDefault
import top.bettercode.summer.web.serializer.annotation.JsonUrl
import java.math.BigDecimal
import java.util.*
import javax.validation.constraints.NotNull

open class DataDicBean {
    @JsonDefault("0")
    var number1: @NotNull BigDecimal? = null

    @JsonBigDecimal(scale = 3)
    var number2: BigDecimal? = null

    @JsonDefault(fieldName = "number2")
    @JsonBigDecimal(scale = 3, stripTrailingZeros = true)
    var number22: BigDecimal? = null

    @JsonBigDecimal(scale = 4, toPlainString = true, percent = true, stripTrailingZeros = true)
    var number3: BigDecimal? = null

    @JsonBigDecimal(scale = 4, toPlainString = true, stripTrailingZeros = true)
    var number4: BigDecimal? = null
    var name: String? = null
    var code: String? = null
    open var intCode: Int? = null

    @Unit
    var price: Long? = null

    @JsonUrl
    var path: String? = null
    var path1: String? = null
    var desc: String? = null
    var paths: MutableList<String>? = null
    var pathArray: Array<String>? = null
    val paths1: List<String?>?
        get() = paths

    open fun getPathArray1(): Array<String>? {
        return pathArray
    }


    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is DataDicBean) {
            return false
        }
        return number1 == other.number1 && number2 == other.number2 && number22 == other.number22 && number3 == other.number3 && number4 == other.number4 && name == other.name && code == other.code && intCode == other.intCode && price == other.price && path == other.path && path1 == other.path1 && desc == other.desc && paths == other.paths && pathArray.contentEquals(other.pathArray)
    }

    override fun hashCode(): Int {
        var result = Objects.hash(number1, number2, number22, number3, number4, name, code, intCode,
                price, path, path1, desc, paths)
        result = 31 * result + pathArray.contentHashCode()
        return result
    }

    override fun toString(): String {
        return "DataDicBean{" +
                "number1=" + number1 +
                ", number2=" + number2 +
                ", number22=" + number22 +
                ", number3=" + number3 +
                ", number4=" + number4 +
                ", name='" + name + '\'' +
                ", code='" + code + '\'' +
                ", intCode=" + intCode +
                ", price=" + price +
                ", path='" + path + '\'' +
                ", path1='" + path1 + '\'' +
                ", desc='" + desc + '\'' +
                ", paths=" + paths +
                ", pathArray=" + pathArray.contentToString() +
                '}'
    }
}